### PR TITLE
Feature/serv 643 advisories

### PR DESCRIPTION
**Pull request by turegjorup**
#### Link to ticket

https://jira.itkdev.dk/browse/SERV-643

#### Description

- Added advisory handler and UI
- Fixed "Integrity constraint violation: 1062 Duplicate entry" errors
- Minor UI styling updates
- Added logo and favicon

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Checklist

- [ ] My code is covered by test cases.
- [ ] My code passes our test (all our tests).
- [ ] My code passes our static analysis suite.
- [ ] My code passes our continuous integration process.

If your code does not pass all the requirements on the checklist you have to add a comment explaining why this change 
should be exempt from the list.

#### Additional comments or questions

If you have any further comments or questions for the reviewer please add them here.

### CODE REVIEW

File: src/Controller/Admin/AdvisoryCrudController.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+//    public function configureFilters(Filters $filters): Filters
+//    {
+//        return $filters
+//            ->add('vendor')
+//            ->add('name')
+//            ->add('type')
+//            ->add('license')
+//        ;
+//    }
```
Comment (jekuaitk) [inline src/Controller/Admin/AdvisoryCrudController.php]: Do we want some filters?

File: src/Repository/AdvisoryRepository.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+//    public function findOneBySomeField($value): ?Advisory
+//    {
+//        return $this->createQueryBuilder('a')
+//            ->andWhere('a.exampleField = :val')
+//            ->setParameter('val', $value)
+//            ->getQuery()
+//            ->getOneOrNullResult()
+//        ;
+//    }
```
Comment (jekuaitk) [inline src/Repository/AdvisoryRepository.php]: Can be deleted

File: src/Service/ModuleVersionFactory.php [hunk 3]
```diff
@@ -12,6 +12,9 @@
 
 class ModuleVersionFactory
 {
+    private array $createdModules = [];
+    private array $createdModuleVersions = [];
```
Comment (jekuaitk) [inline src/Service/ModuleVersionFactory.php:16]: Is something ever added to these?

File: src/Service/ModuleVersionFactory.php [hunk 4]
```diff
@@ -23,41 +26,75 @@ public function setModuleVersions(Installation $installation, object $installedM
...
+            foreach ($this->createdModules as $createdModule) {
+                if ($name === $createdModule->getName() && $package === $createdModule->getPackage()) {
+                    $module = $createdModule;
                 }
-                $module->setEnabled('Enabled' === $installed->status);
             }
+        }
+
+        if (null === $module) {
+            $module = new Module();
```
Comment (jekuaitk) [inline src/Service/ModuleVersionFactory.php:64]: Should this be added to `$createdModules`?

File: src/Service/ModuleVersionFactory.php [hunk 5]
```diff
@@ -23,41 +26,75 @@ public function setModuleVersions(Installation $installation, object $installedM
...
+            foreach ($this->createdModuleVersions as $createdModuleVersion) {
+                if ($module->getId() === $createdModuleVersion->getModule()->getId() && $version === $createdModuleVersion->getVersion()) {
+                    $moduleVersion = $createdModuleVersion;
+                }
             }
+        }
 
-            $moduleVersion->setVersion($installed->version);
+        if (null === $moduleVersion) {
+            $moduleVersion = new ModuleVersion();
```
Comment (jekuaitk) [inline src/Service/ModuleVersionFactory.php:93]: Should this be added to `$createdModuleVersions`?

File: src/Service/ModuleVersionFactory.php (hunk 3 above)
Comment (turegjorup) [reply src/Service/ModuleVersionFactory.php:16]: Fixed

File: src/Service/ModuleVersionFactory.php (hunk 5 above)
Comment (turegjorup) [reply src/Service/ModuleVersionFactory.php:93]: Fixed

**Review by jekuaitk — APPROVED** [review]
Looks good to me!

**Pull request merged**